src/realm/util/ez_websocket.cpp

```diff
@@ -0,0 +1,345 @@
+#include <realm/util/ez_websocket.hpp>
+
+#include <realm/util/websocket.hpp>
+#include <realm/util/network.hpp>
+#include <realm/util/network_ssl.hpp>
+
+namespace realm::util::websocket {
+
+namespace {
+class EZSocketImpl final : public EZSocket, public websocket::Config {
```

Comment:
Feels odd that this inherits from `webSocket::Config`... I'm assuming this is an artefact from the old implementation?

Reply:
Yeah. The old Config type was a weird combination of EZConfig, EZObserver, and an injectable low level socket abstraction. ClientImpl::Connection used to inherit from websocket::Config. Part of the change here is teasing apart the parts of Config, and part is moving the low level socket inside the websocket abstraction rather than sitting outside of it.